tests/drivers/pwm/pwm_api/src/main.c

```diff
@@ -8,15 +8,21 @@
 #include <zephyr.h>
 #include <ztest.h>
 
+const struct device *get_pwm_device(void);
 void test_pwm_usec(void);
 void test_pwm_cycle(void);
 void test_pwm_nsec(void);
 
 void test_main(void)
 {
+	const struct device *dev = get_pwm_device();
+
+	zassert_not_null(dev, "Cannot get PWM device");
+	k_object_access_grant(dev, k_current_get());
+
 	ztest_test_suite(pwm_basic_test,
-			 ztest_unit_test(test_pwm_usec),
-			 ztest_unit_test(test_pwm_nsec),
-			 ztest_unit_test(test_pwm_cycle));
+			 ztest_user_unit_test(test_pwm_usec),
```

Comment:
Isn't it common to test both supervisor and userspace, rather than just userspace?  I'm used to seeing two test cases, one with and one without, with the code shared.

Reply:
@pabigot Thank you for taking a look at this. I would think testing in user space would be enough? Do you have in-tree examples of the above approach?

Reply:
None that are obvious; I'm just used to seeing tests that verify behavior from both privileged threads and userspace threads.  I have no use for userspace myself, but I guess supervisory mode can be assumed to work if userspace does unless there's something odd going on with the wrappers.